### PR TITLE
KHWE-36/37: Connecting share button AND Legal info with firebase

### PR DESCRIPTION
**Pull request by pavlobaranchuk**

### CODE REVIEW

File: src/components/DotsMenu/DotsMenu.js [hunk 1]
```diff
@@ -33,7 +33,7 @@ export class DotsMenuComponent {
     const activeMenuItem = e.target.closest(".dots-menu__item");
     if (activeMenuItem) {
       const id = parseInt(activeMenuItem.dataset.id, 10);
-      this.props.items[id].handler();
+      this.props.items[id].handler(activeMenuItem);
```
Comment (ihor4568) [inline src/components/DotsMenu/DotsMenu.js]: you should not pass an element here. You can pass a value. Other components should not have access to internals of this component

File: src/components/Main/Main.html [hunk 2]
```diff
@@ -20,7 +20,7 @@ <h3 class="mdc-drawer__title main__user">
           <i class="material-icons mdc-list-item__graphic" aria-hidden="true">person_outline</i>
           <span class="mdc-list-item__text">Artists</span>
         </a>
-        <hr class="mdc-list-divider">
+        <hr class="mdc-list-divider" />
```
Comment (ihor4568) [inline src/components/Main/Main.html]: there are no self-closing tags in html spec - it is from xml/xhtml. In html there are just tags that do not have closing tag - like input, hr, img -- so remove that /

File: src/components/Main/Main.js [hunk 3]
```diff
@@ -199,7 +200,13 @@ export class MainComponent {
     });
     this.header.mount();
 
+    this.licenseDialogComponent = new LicenseDialogComponent(
+      this.licenseDialog
+    );
+    this.licenseDialogComponent.mount();
+
     this.player = new MediaPlayerComponent(this.playerPoint, {
+      licenseDialogComponent: this.licenseDialogComponent,
```
Comment (ihor4568) [inline src/components/Main/Main.js]: please do not pass component as a property to other component

File: src/components/Main/Main.js [hunk 4]
```diff
@@ -214,7 +221,8 @@ export class MainComponent {
     this.about = new AboutComponent(this.mainContentPoint);
     this.table = new MySongsComponent(this.mainContentPoint, {
       onSongPlay: this.handleSongPlay.bind(this),
-      onSongStop: this.handleSongStop.bind(this)
+      onSongStop: this.handleSongStop.bind(this),
+      licenseDialogComponent: this.licenseDialogComponent
```
Comment (ihor4568) [inline src/components/Main/Main.js]: please do not pass component as a property to other component

File: src/components/ShareView/ShareView.js [hunk 5]
```diff
@@ -1,16 +1,49 @@
...
+import { MusicService } from "../../services/MusicService";
 
 export class ShareViewComponent {
   constructor(mountPoint, props = {}) {
     this.mountPoint = mountPoint;
     this.props = props;
+    this.title = {};
+  }
+
+  fetchAuthors(authorsIds) {
```
Comment (ihor4568) [inline src/components/ShareView/ShareView.js]: fetchAuthors, fetchInfoBySong, fetchSongs... these methods are very similar to those in mymusic. Could you please extract them to the MusicService and reuse here and there?

File: src/components/Main/Main.html (hunk 2 above)
Comment (pavlobaranchuk) [reply src/components/Main/Main.html]: I didn't put this sign, it is auto generated oNsave. I will try to figure it out who added it...

File: src/components/ShareView/ShareView.js (hunk 5 above)
Comment (pavlobaranchuk) [reply src/components/ShareView/ShareView.js]: Done

**Pull request closed without merging**